third_party/terraform/utils/provider.go.erb

```diff
@@ -435,21 +431,25 @@ end     # products.each do
 		)
 }
 
-func providerConfigure(d *schema.ResourceData, p *schema.Provider, terraformVersion string) (interface{}, error) {
+func providerConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Provider) (interface{}, diag.Diagnostics) {
 	config := Config{
 		Project:             d.Get("project").(string),
 		Region:              d.Get("region").(string),
 		Zone:                d.Get("zone").(string),
 		UserProjectOverride: d.Get("user_project_override").(bool),
 		BillingProject:      d.Get("billing_project").(string),
-		terraformVersion:    terraformVersion,
+        <% if version.nil? || version == 'ga' -%>
+        userAgent: p.UserAgent("terraform-provider-google", version.ProviderVersion),
```

Comment:
Can you post a before/after value for the user agent string?

Reply:
User-Agent: Terraform/0.12.29 (+https://www.terraform.io) Terraform-Plugin-SDK/2.0.3 terraform-provider-google-beta/acc

Reply:
Thanks!